pkg/topology/topology.go

```diff
@@ -300,6 +300,12 @@ func findSysFsDevice(devType string, major, minor int64) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("failed to get realpath for %s: %w", devPath, err)
 	}
+	if sysRoot != "" {
+		realDevPath = strings.TrimPrefix(realDevPath, sysRoot)
```

Comment:
should we better to check that if the `realDevPath` prefixed with `sysRoot`, instead of just checking if `sysRoot` is not empty?

Reply:
As a side note, in `SetSysRoot` we can use `filepath.Clean()` to make sure that the value is correct, so operations with those paths will be more predictable. e.g. in line 181.

Reply:
Updated to check prefix against cleaned sysroot.

Reply:
Path returned by `filepath.Clean()` is promised not to end with "/" except for in the case of the root directory. Therefore, if user calls `SetSysRoot("/")`, the latest change seems to make it possible that `findSysFsDevice()` returns paths that do not start with "/". I'm not sure if this will cause issues, and whether or not it is possible to call `SetSysRoot("/")`, but at least it is more possible than it was before.

Reply:
Updated to 1) only use implicit "/" as sysroot and 2) always resolve relative sysroot to absolute one, panicking if this fails. Added a few basic test cases for `SetSysRoot()`.